### PR TITLE
Log in and log out 

### PR DESCRIPTION
**Pull request by edgarmagana**
In this PR, log in page and log out functionality was  added to tickter proyect, also there's  a change in table tickets in order to fix  an issue found by Marcelo.

### CODE REVIEW

File: ui/src/js/appViewModel.js [hunk 1]
```diff
@@ -1,129 +1,187 @@
...
+                self.login.user.subscribe(() => {
+                    self.login.isError(false);
+                });
+                self.login.password.subscribe(() => {
+                    self.login.isError(false);
+                });
+            });
+        };
+
+        self.userInformation = () => {
```
Comment (reymus) [inline ui/src/js/appViewModel.js:162]: This is not a valid function name. Use an action name for functions. Never a noun.

File: ui/src/js/appViewModel.js [hunk 2]
```diff
@@ -1,129 +1,187 @@
...
-    // Add a close listener so we can move focus back to the toggle button when the drawer closes
-    document.getElementById('navDrawer').addEventListener("ojclose", function() {
-      document.getElementById('drawerToggleButton').focus();
-    });
-  }
-
-  return new AppViewModel();
-});
+        };
+        self.loginFormatBody = (email, password) => {
```
Comment (reymus) [inline ui/src/js/appViewModel.js:177]: Modify this to be called `getLoginPayload`  and remove the parameters.

File: ui/src/js/appViewModel.js (hunk 1 above)
Comment (reymus) [reply ui/src/js/appViewModel.js:162]: Create a new observable in `appViewModel` called `authenticatedUser` and store the full user object in it. Create an additional function to retrieve the full name of the authenticated user.

File: ui/src/js/appViewModel.js [hunk 3]
```diff
@@ -1,129 +1,187 @@
...
+            errorMessage: ko.observable(""),
+            emailPatternValidator: ko.observableArray([
+                new AsyncRegExpValidator({
+                    pattern: "[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*",
+                    hint: "enter a valid email format",
+                    messageDetail: "Not a valid email format"
+                })
+            ])
+        }
+        self.sendLoginForm = async() => {
```
Comment (reymus) [inline ui/src/js/appViewModel.js:145]: Somewhere around here, add a check to see if the user is logged in, if so, set the `authenticatedUser` (see note below) observable to the jwt payload.

File: ui/src/js/utils/cache.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+define([], function() {
```
Comment (reymus) [inline ui/src/js/utils/cache.js:1]: Document this module. Add comments regarding the motivation and a comment for each function.
`Cache` should be a constant (use `const`).

File: ui/src/js/utils/decoder.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+define([], function(require, factory) {
+    'use strict';
```
Comment (reymus) [inline ui/src/js/utils/decoder.js:2]: Document, same as cache. 

Remove `require` and `factory` (make sure you don't leave evidence of your copy-pasted code)

File: ui/src/js/utils/decoder.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    let Decoder = {
+        parseJwt: function(token) {
+            var base64Url = token.split('.')[1];
```
Comment (reymus) [inline ui/src/js/utils/decoder.js:6]: Comment on what the eff is doing this function.

Comment (reymus) [reply ui/src/js/utils/decoder.js:6]: Be consistent. Use `let` instead of `var`.

Comment (edgarmagana) [reply ui/src/js/utils/decoder.js:6]: done

File: ui/src/js/utils/decoder.js (hunk 5 above)
Comment (edgarmagana) [reply ui/src/js/utils/decoder.js:2]: done

File: ui/src/js/utils/cache.js (hunk 4 above)
Comment (edgarmagana) [reply ui/src/js/utils/cache.js:1]: done

File: ui/src/js/appViewModel.js (hunk 2 above)
Comment (edgarmagana) [reply ui/src/js/appViewModel.js:177]: done

File: ui/src/js/appViewModel.js (hunk 1 above)
Comment (edgarmagana) [reply ui/src/js/appViewModel.js:162]: done

**Pull request merged**